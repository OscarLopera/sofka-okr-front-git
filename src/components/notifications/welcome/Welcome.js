import React, { useState } from 'react';
import Modal from 'react-modal';
import { auth } from '../../../functions/firebaseAuth';
import '../../../assets/styles/notifications/notificationsWindows.scss';
import { useAuthState } from "react-firebase-hooks/auth";

const nameUser =  auth;

const Welcome = () => {
    const [modalIsOpen, setModalIsOpen] = useState(true)
    const [user] = useAuthState(auth);
    const customStyles = {
        content: {
            top: '20%',
            left: '50%',
            right: 'auto',
            bottom: 'auto',
            marginRight: '-50%',
            width: '600px',
            transform: 'translate(-40%, -10%)',
            padding: '0',
        },
    };

    return (
        <div id="root">
            <Modal isOpen={modalIsOpen}
                onRequestClose={() => setModalIsOpen(false)}
                style={customStyles}>
                <header className='headerNotificationWindows' id='headerNotificationWindows'></header>
                <div className='bodyNotificationWindows' id='bodyNotificationWindows'>
                    <h1 className='h1NotificationWindows' id='h1NotificationWindows'> Â¡Bienvenido {user? nameUser.currentUser.displayName :""}! </h1>
                    <p className='pNotificationWindows' id='pNotificationWindows'>"Estamos felices de tenerte en nuesto Dashboard, esperamos que podamos aprender de tus aportes"</p>
                </div>
                <div className='imageNotificationWindows' id='imageNotificationWindows'>
                    <img src="https://www.sofka.com.co/wp-content/uploads/2020/08/sofka-logo-gradient-white.png" alt="logo-sofka"></img>
                </div>
                <footer className='footerNotificationWindows' id='footerNotificationWindows'></footer>
            </Modal>
        </div>
    )
}

export default Welcome;
