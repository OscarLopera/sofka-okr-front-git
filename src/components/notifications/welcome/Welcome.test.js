import React from 'react';
import Modal from 'react-modal';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Welcome from './Welcome'
import sinon from 'sinon'

configure({ adapter: new Adapter() });

it('show content of modal', ()=>{
    const spy  = sinon.spy();
    const wrapper = shallow(<Welcome onTest={spy}></Welcome>)
    expect(spy.called)
    wrapper.unmount();
})

it('renders react-modal', () => {
    const wrapper = shallow(<Welcome />);
    expect(wrapper.find(Modal)).toHaveLength(1);
});

it('opens modal when button is clicked', () => {
    const wrapper = shallow(<Welcome />);
    expect(wrapper.find(Modal).prop('isOpen')).toBe(true);
});

it('renders childen when modal is open', () => {
    const wrapper = shallow(<Welcome>modal content</Welcome>);
    expect(wrapper.find(Modal).prop('children')).toMatchSnapshot();
});

