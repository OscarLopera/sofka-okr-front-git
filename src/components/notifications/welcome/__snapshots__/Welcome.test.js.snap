// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders childen when modal is open 1`] = `
Array [
  <header
    className="headerNotificationWindows"
    id="headerNotificationWindows"
  />,
  <div
    className="bodyNotificationWindows"
    id="bodyNotificationWindows"
  >
    <h1
      className="h1NotificationWindows"
      id="h1NotificationWindows"
    >
       Â¡Bienvenido 
      
      ! 
    </h1>
    <p
      className="pNotificationWindows"
      id="pNotificationWindows"
    >
      "Estamos felices de tenerte en nuesto Dashboard, esperamos que podamos aprender de tus aportes"
    </p>
  </div>,
  <div
    className="imageNotificationWindows"
    id="imageNotificationWindows"
  >
    <img
      alt="logo-sofka"
      src="https://www.sofka.com.co/wp-content/uploads/2020/08/sofka-logo-gradient-white.png"
    />
  </div>,
  <footer
    className="footerNotificationWindows"
    id="footerNotificationWindows"
  />,
]
`;
